hazelcast/include/hazelcast/client/DistributedObject.h

```diff
@@ -64,7 +65,7 @@ namespace hazelcast {
         * @see TransactionalSet
         * @see TransactionalList
         */
-        class HAZELCAST_API DistributedObject : public IDistributedObject {
+        class HAZELCAST_API DistributedObject : public virtual IDistributedObject {
```

Comment:
Asking because I am curious. Can you describe why we need this ? What did it solve ?

Reply:
Reason is this: ClientPNCounterProxy inherits two classes (crdt::pncounter::PNCounter and ProxyImpl) . crdt::pncounter::PNCounter is only pure virtual interface. PNCounter extends the interface IDistributedObject and ProxyImpl also implements IDistributedObject. When the derived class inherits the base class through more than one class then we need to make the base class as virtual.